packages/app/src/util/credential.ts

```diff
@@ -0,0 +1,205 @@
+import { VerifiableCredential, Issuer, CredentialsProof, Subject, Status, ClaimValue } from "@src/types";
+
+/**
+ * Attempts to parse a JSON string into a VerifiableCredential object.
+ * @param obj A JSON string representing a VerifiableCredential.
+ * @returns An object representing a VerifiableCredential. Null if the object is not a valid VerifiableCredential.
+ */
+export function parseVerifiableCredentialFromJson(json: string): VerifiableCredential | null {
+  if (!json || typeof json !== "string") {
+    return null;
+  }
+
+  let obj;
+  try {
+    obj = JSON.parse(json) as VerifiableCredential;
+  } catch (e) {
+    return null;
+  }
+
+  if (
+    obj.issuanceDate &&
```

Comment:
we have `yup` schema validation so you can write rules without adding custom validation mechanism.
This is related to all functions in this file.

Reply:
updated to use yup schema validation